src/improvements/doc/TASK_CREATION_GUIDE.md

```diff
@@ -0,0 +1,196 @@
+# Task Creation Guide
+
+This guide walks through the process of creating new tasks using existing templates, including network-specific considerations and best practices.
+
+## Quick Start
+
+To scaffold a new task using an existing template:
+
+```bash
+cd src/improvements/
```

Comment:
Again maybe some TODOs here. Or possibly just a higher level tracking issue to make sure these are revisited.